### PR TITLE
Use RoundTripper-like interface for interceptors.

### LINKED ISSUE #269: Use RoundTripper-like interface for interceptors
Tracking bug for modify interceptors to model after [Golang HTTP RoundTrippers](https://godoc.org/net/http#RoundTripper)

Currently:

https://github.com/tektoncd/triggers/blob/3c645a9f50abb3a59cbd5613de8cdc9ee9d9a39f/pkg/interceptors/interceptors.go#L27-L30

Proposed:

```go
type Interceptor interface {
	ExecuteTrigger(req *http.Request, trigger *triggersv1.EventListenerTrigger, eventID string) (*http.Response, error)
}
```

### PR DESCRIPTION
**Pull request by wlynch**
# Changes

This simplifies the interceptor interface to look like
http.RoundTripper, having interceptors take in an http.Request and
returning an http.Response.

The only substantial behavior change this makes is gives webhook interceptors
the ability to respond back with HTTP headers.

Fixes #269.

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [x] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [x] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/triggers/blob/master/CONTRIBUTING.md) for more details._

### CODE REVIEW

File: pkg/interceptors/github/github_test.go [hunk 1]
```diff
@@ -218,11 +223,19 @@ func TestInterceptor_ExecuteTrigger_Signature(t *testing.T) {
...
-			if (err != nil) != tt.wantErr {
-				t.Errorf("Interceptor.ExecuteTrigger() error = %v, wantErr %v", err, tt.wantErr)
+			resp, err := w.ExecuteTrigger(request)
+			if err != nil {
+				if !tt.wantErr {
+					t.Errorf("Interceptor.ExecuteTrigger() error = %v, wantErr %v", err, tt.wantErr)
+				}
 				return
 			}
+			fmt.Println(resp, err)
```
Comment (dibyom) [inline pkg/interceptors/github/github_test.go]: Intentional?

Comment (wlynch) [reply pkg/interceptors/github/github_test.go]: Whoops. Removed.

**Pull request merged**